### ISSUE TITLE
Antlr 4.9.4 preview: errors in antlr/antlr4 and v/

### ISSUE THREAD

**Issue by kaby76** [antlr/grammars-v4#2481]
I've been updating the Antlr4 tool for the Go target and decided to test out the new tool on grammars-v4/ as this is a more extensive test that the unit tests for the Antlr4 tool.

* error(153): ANTLRv3Lexer.g4:113:0: rule DOUBLE_ANGLE_STRING_LITERAL contains a closure with at least one alternative that can match an empty string
* error(186): V.g4:670:0: rule elementList contains a closure with at least one alternative that can match EOF

**Comment by KvanTTT** [comment]
Thanks for testing 4.9.4 (actually 4.10) on the entire grammars repository. I just wanted to start a discussion about that. Also, `caseInsensitive` option should be tested (I've already started it).

As I see they the errors are completely valid. But in V.g4 it probably should be several ` contains a closure with at least one alternative that can match EOF` errors (see rules with `eos` under closure).

**Comment by kaby76** [comment]
@KvanTTT I added a check to trgen to fail on caseInsensitive = true but I just forgot what you were working on. Sorry. I'll fix the check so that it doesn't crash on the value. I'll fix antlr/antlr3 and v soon when I can get past https://github.com/antlr/antlr4/pull/3486

**Comment by KvanTTT** [comment]
> I added a check to trgen to fail on caseInsensitive = true but I just forgot what you were working on. Sorry. I'll fix the check so that it doesn't crash on the value

Yes, I've also transformed some grammar with fragments `TOKEN: T O K E N` -> `TOKEN: `TOKEN`;`, but have not completed yet.

**Comment by kaby76** [comment]
The v/ grammar here is not very good. It does not parse most files in the ["vlib"](https://github.com/vlang/v/tree/master/vlib) runtime library.

There are various discussions about whether someone could write down the EBNF for V. But, likely it'll never be done ([1](https://github.com/vlang/v/issues/1825#issuecomment-527363780)).

Of course, there is a parser for V. It is hand-written code (why would it be any different for V as with every other language in our miserable profession) ([2](https://github.com/vlang/v/blob/master/vlib/v/parser/parser.v)). It's noted that "[u]nlike many other languages, V is not going to be always changing" ([3](https://github.com/vlang/v#stability-guarantee-and-future-changes)), but there have been 13 changes in parser.v over the last month alone ([4](https://github.com/vlang/v/commits/master/vlib/v/parser/parser.v)).

The VSCode extension for V ([5](https://github.com/vlang/vscode-vlang)) is a TextMate implementation, so at best it gives only the lexical structure, in the great EBNF syntax JSON at that.

I recommend redoing the grammar by scraping it from the Tree Sitter grammar for V ([6](https://github.com/vlang/vls/blob/master/tree_sitter_v/src/grammar.json) or [7](https://github.com/vlang/vls/blob/master/tree_sitter_v/grammar.js)) within the LSP server code. It is being maintained. 
[v-raw-scrapped.txt](https://github.com/antlr/grammars-v4/files/7914674/v-raw-scrapped.txt) However, I don't know if one would end up right back here: a grammar that isn't very good.

**Comment by kaby76** [comment]
I've reworked the convertor from tree-sitter to pseudo-Antlr4 grammars. As it turns out, one can't really use the grammar.js as input for the conversion: Tree-sitter performs a partial evaluation of the .js code. The table used in the V grammar for expressions must be converted by tree-sitter itself. The converter must use the output from tree-sitter, the grammar.json file. The raw grammar for V is here: 
[grammar.txt](https://github.com/antlr/grammars-v4/files/7938035/grammar.txt)
